tests/test_mab.py

```diff
@@ -1767,3 +1767,25 @@ def test_pickle_fitted_reproducibile_expectations(self):
                 self.assertListEqual(pe1, pe2)
 
         os.remove('mab.pkl')
+
+    def test_cold_arms(self):
+        _, mab = self.predict(arms=[1, 2, 3],
+                              decisions=[1, 1, 1, 1, 2, 2, 2, 1, 2, 1],
+                              rewards=[0, 1, 1, 0, 1, 0, 1, 1, 1, 1],
+                              learning_policy=LearningPolicy.LinGreedy(epsilon=0.25),
+                              context_history=[[0, 1, 2, 3, 5], [1, 1, 1, 1, 1], [0, 0, 1, 0, 0],
+                                               [0, 2, 2, 3, 5], [1, 3, 1, 1, 1], [0, 0, 0, 0, 0],
+                                               [0, 1, 4, 3, 5], [0, 1, 2, 4, 5], [1, 2, 1, 1, 3],
+                                               [0, 2, 1, 0, 0]],
+                              contexts=[[0, 1, 2, 3, 5], [1, 1, 1, 1, 1]],
+                              seed=123456,
+                              num_run=4,
+                              is_predict=True)
+
+        # Before warm start
+        self.assertEqual(mab._imp.trained_arms, [1, 2])
+        self.assertSetEqual(mab.cold_arms, {3})
+
+        # Warm start
+        mab.warm_start(arm_to_features={1: [0, 1], 2: [0, 0], 3: [0.5, 0.5]}, distance_quantile=0.5)
```

Comment:
Am I reading this right: 

So we already have a "warm_start()" function to copy models based on features and quantile. (was that the other PR you had?) 

And now, here, you are adding the ability to for the system to store and track "cold_arms". 

Together, a user can "warm_start" things, and at any point, ask what is "cold". 

Design question; any pros/cons if this was not storing a set of cold_arms only but instead sth like "arm_to_status" where status can be "cold", "trained", "warm", "warm+tuned", etc.

Reply:
As Bernard said, the previous PR was on updating warm start to account for multiple calls to `warm_start`.

I really like the `arm_to_status` idea! We could use that to consolidate `trained_arms` and `cold_arms`.

```python
arm_to_status = {"a": "cold", "b", "warm", "c": "trained"}
```

I'm a bit skeptical of also merging in `cold_arm_to_warm_arm`, because then we'd have to incorporate that info into the `arm_to_status` dict, which could make it a little more confusing. However, I really like the single source of truth with one object, so I'd prefer to implement that as well. Consider something like this:

```python
arm_to_status = {"a": "cold", "b": ("warm", "a"), "c": "trained"}

Reply:
I am like the design considerations.. just wanted to mention cold, warm, trained, and warm+tuned (if possible). There will be two things critical 1) setting these correctly, and 2) (harder) to maintain them correctly going forward. For instance fit() operation now has to switch a warm arm to warm+tuned when it sees the first data point and so on. 

I am OK with "status" as the field name, as in, arm_to_status. I don't have any better suggestion. My only consideration is if/when that is used from outside, gotta make sure if it something that is hard to confuse with arm_to_is_active (like I did tripped last time)